MyOnboarding/test/MyOnboardingApp.Validation.Tests/Validators/ValidatorTests.cs

```diff
@@ -0,0 +1,112 @@
+
+            var result = _validator.Validate(testItem);
+
+            criterion1.Received(1).Validate(testItem);
+            criterion2.Received(1).Validate(testItem);
+            Assert.That(result.Errors, Is.EqualTo(errors));
+            Assert.That(result.WasOperationSuccessful, Is.EqualTo(expectedResult.WasOperationSuccessful));
+        }
+
+
+        [Test]
+        public void Validate_AllCriteriaFulfilled_ReturnsCorrectItemWithoutErrors()
+        {
+            var criterion1 = Substitute.For<IValidationCriterion<TodoListItem>>();
+            var criterion2 = Substitute.For<IValidationCriterion<TodoListItem>>();
+            _validator = new Validator<TodoListItem>(new[] { criterion1, criterion2 });
+            var testItem = new TodoListItem
+            {
+                Id = new Guid("00000000-0000-0000-0000-000000000002"),
+                Text = "test",
+                CreationTime = new DateTime(2010, 01, 01),
+                LastUpdateTime = new DateTime(2010, 01, 01)
+            };
+
+            criterion1.Validate(testItem).Returns(new List<Error>());
+            criterion2.Validate(testItem).Returns(new List<Error>());
+            var expectedResult = ItemVariantsFactory.CreateItemVariants(testItem, new List<Error>()).ItemWithErrors;
+
+            var result = _validator.Validate(testItem);
+
+            criterion1.Received(1).Validate(testItem);
+            criterion2.Received(1).Validate(testItem);
+            Assert.That(result.WasOperationSuccessful, Is.EqualTo(expectedResult.WasOperationSuccessful));
+            Assert.That(result.Item, Is.EqualTo(expectedResult.Item));
+            Assert.That(result.Errors, Is.EqualTo(expectedResult.Errors));
+        }
+
+
+        [Test]
+        public void Validate_OneCriterionNotFulfilled_ReturnsCorrectItemWithoutErrors()
+        {
+            var criterion1 = Substitute.For<IValidationCriterion<TodoListItem>>();
+            var criterion2 = Substitute.For<IValidationCriterion<TodoListItem>>();
+            _validator = new Validator<TodoListItem>(new[] { criterion1, criterion2 });
+            var testItem = new TodoListItem
+            {
+                Id = new Guid("00000000-0000-0000-0000-000000000002"),
+                Text = "test",
+                CreationTime = new DateTime(2010, 01, 01),
+                LastUpdateTime = new DateTime(2010, 01, 01)
+            };
+            var criterion1Errors = new List<Error>
+            {
+                new Error(ErrorCode.DataValidationError, "validation message", "validation.location")
+            };
+            // ReSharper disable once CollectionNeverUpdated.Local
+            var criterion2Errors = new List<Error>();
+            criterion1.Validate(testItem).Returns(criterion1Errors);
+            criterion2.Validate(testItem).Returns(criterion2Errors);
+            var errors = new List<Error>(criterion1Errors.Concat(criterion2Errors));
```

Comment:
R# warnings are always worth consideration… You disable them too often too freely 😉 
Is this code what you mean?
```suggestion
            var errors = Enumerable
                .Empty<Error>()
                .Union(new [] { new Error(ErrorCode.DataValidationError, "validation message", "validation.location") })
                .Union(criterion1.Validate(testItem))
                .Union(criterion2.Validate(testItem))
                .ToArray();
```
or more conventionally:
```suggestion
            var errors = new List<Error>();
            var preExistingErrors = new []
            {
                new Error(ErrorCode.DataValidationError, "validation message", "validation.location")
            };

            var criterion1Errors = criterion1.Validate(testItem);
            var criterion2Errors = criterion2.Validate(testItem);

            errors.AddRange(preExistingErrors);
            errors.AddRange(criterion1Errors);
            errors.AddRange(criterion2Errors);
```
(there are plenty variants of course)